sympy/matrices/random.py

```diff
@@ -0,0 +1,2248 @@
+            _set_value(obj, col, col, 0)
+        else:
+            # add scalar multiple to another
+            _set_value(obj, row, col, value)
+    elif len(scalar_set) == 3:
+        _set_value(obj, row, row, value)
+        y, v, u = scalar_set
+        _set_value(obj, row, col, y)
+        _set_value(obj, col, row, -v)
+        _set_value(obj, col, col, u)
+    else:
+        msg = "either no, one or three additional scalar_set arguments " \
+              "may be supplied, but not %i"
+        raise ValueError(msg % len(scalar_set))
+    return obj
+
+
+# === fundamental matrix functions ===
+
+
+def complex_to_real(mat=None):
+    r""" returns real version of a complex matrix
+
+    Explanation
+    ===========
+    Any complex vector space $V$ with basis $(b_1, \dots, b_n)$
+    can be seen as a real vector space $V^{prime}$ with basis
+    $(b_1, b_1 \ i, \dots, b_n, b_n \ i)$
+    of dimension $2 \ n$.
+
+    Since multiplication with complex number is linear on $V$
+    resp. $V^{\prime}$,
+    any complex homomorphism (i.e. complex matrix $A$) on $V$
+    becomes a real homomorphism (i.e. matrix $A^{\prime}$) on $V^{\prime}$.
+
+    This transforms a complex matrix $A$
+    to the corresponding real matrix $A^{\prime}$.
+
+    Examples
+    ========
+    >>> from sympy import Matrix, I
+    >>> from sympy.abc import a, b
+    >>> from sympy.matrices.random import complex_to_real
+
+    >>> z = a + b * I
+    >>> A = Matrix([[z]])
+    >>> A
+    Matrix([[a + I*b]])
+
+    >>> complex_to_real(A)
+    Matrix([
+    [ re(a) - im(b), re(b) + im(a)],
+    [-re(b) - im(a), re(a) - im(b)]])
+
+    >>> A = Matrix([[z, 0],[1, I]])
+    >>> A
+    Matrix([
+    [a + I*b, 0],
+    [      1, I]])
+    >>> complex_to_real(A)
+    Matrix([
+    [ re(a) - im(b), re(b) + im(a),  0, 0],
+    [-re(b) - im(a), re(a) - im(b),  0, 0],
+    [             1,             0,  0, 1],
+    [             0,             1, -1, 0]])
+
+    Parameters
+    ==========
+    mat : Matrix
+        a complex matrix
+
+    """
+    dim = _shape(mat)[0]
+    mat = mat or identity_matrix(dim)
+    obj = super_elementary_matrix(2 * dim)
+    for i in range(dim):
+        for j in range(dim):
+            z_value = _get_value(mat, i, j)
+            a, b = _re(z_value), _im(z_value)
+            _set_value(obj, 2 * i, 2 * j, a)
+            _set_value(obj, 2 * i, 2 * j + 1, b)
+            _set_value(obj, 2 * i + 1, 2 * j, -b)
+            _set_value(obj, 2 * i + 1, 2 * j + 1, a)
+    return obj
+
+
+def regular_to_singular(mat, rank=None):
+    r""" build matrix with linear combination of matrix columns to meet rank
+
+    Explanation
+    ===========
+    Let $I$ be the $n \times n$ identity_matrix matrix
+    and $D$ an diagonal matrix with only zero and one entries
+    of rank $r$ (i.e. all entries but $n-r$ diagonal entries are zero).
+
+    If a full rank matrix $A$ is given, then
+
+    .. math::
+        B = D * (I + A * (I-D)) = D + D * A * (I-D)
```

Comment:
Should this be LaTeX?

Reply:
As suggested, use now `\mathbf` for matrices in LaTeX - and `\cdot` instead of *